commit 0475c3655e6ebd1d6d6f0e705eba97fce39a08e3
Author: Hyunchul Lee <hyc.lee@gmail.com>
Date:   Tue Jun 8 23:53:14 2021 +0900

    cifs: decoding negTokenInit with generic ASN1 decoder
    
    Decode negTokenInit with lib/asn1_decoder. For that,
    add OIDs in linux/oid_registry.h and a negTokenInit
    ASN1 file, "spnego_negtokeninit.asn1".
    And define decoder's callback functions, which
    are the gssapi_this_mech for checking SPENGO oid and
    the neg_token_init_mech_type for getting authentication
    mechanisms supported by a server.
    
    Signed-off-by: Hyunchul Lee <hyc.lee@gmail.com>
    Reviewed-by: Aurelien Aptel <aaptel@suse.com>
    Reviewed-by: Paulo Alcantara (SUSE) <pc@cjr.nz>
    Signed-off-by: Steve French <stfrench@microsoft.com>

diff --git a/fs/cifs/Kconfig b/fs/cifs/Kconfig
index 55bc57e9f812..7364950a9ef4 100644
--- a/fs/cifs/Kconfig
+++ b/fs/cifs/Kconfig
@@ -19,6 +19,8 @@ config CIFS
 	select CRYPTO_LIB_DES
 	select KEYS
 	select DNS_RESOLVER
+	select ASN1
+	select OID_REGISTRY
 	help
 	  This is the client VFS module for the SMB3 family of NAS protocols,
 	  (including support for the most recent, most secure dialect SMB3.1.1)